### PR TITLE
Generate custome proxy

### PR DESCRIPTION
**Pull request by BiancaIalangi**

### CODE REVIEW

File: framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs [hunk 1]
```diff
@@ -315,7 +388,7 @@ impl ScConfig {
                 wasm_crate_name,
                 abi: original_abi.clone(),
             }],
-            proxy_configs: Vec::new(),
+            proxy_configs: HashMap::new(),
```
Comment (andrei-marinica) [inline framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs:391]: Proxies still form a Vec, there is no key

File: framework/meta-lib/src/contract/sc_config/sc_config_model.rs [hunk 2]
```diff
@@ -18,7 +20,7 @@ pub const SC_CONFIG_FILE_NAMES: &[&str] = &["sc-config.toml", "multicontract.tom
 pub struct ScConfig {
     pub default_contract_config_name: String,
     pub contracts: Vec<ContractVariant>,
-    pub proxy_configs: Vec<ProxyConfigSerde>,
+    pub proxy_configs: HashMap<ProxyConfigSerde, ContractVariant>,
```
Comment (andrei-marinica) [inline framework/meta-lib/src/contract/sc_config/sc_config_model.rs:23]: It's still a Vec

File: framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs [hunk 3]
```diff
@@ -253,40 +254,112 @@ fn validate_contract_variants(contracts: &[ContractVariant]) {
...
+    set_main_contract_flag(&mut contracts, &config.settings.main);
+    validate_contract_variants(&contracts);
+
+    contracts
+}
+
+fn process_proxy_contracts(
+    config: &ScConfigSerde,
+    original_abi: &ContractAbi,
+) -> HashMap<ProxyConfigSerde, ContractVariant> {
```
Comment (andrei-marinica) [inline framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs:287]: Should be a Vec

Comment (andrei-marinica) [reply framework/meta-lib/src/contract/sc_config/contract_variant_builder.rs:287]: The config should not be the key here, it should be part of the proxy config
You need to add the ContractVariant to the ProxyConfig

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — DISMISSED** [review]

**Review by mihaicalinluca — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**